integrations/virtiofs/src/filesystem_message.rs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use vm_memory::ByteValued;
+
+/// Opcode represents the filesystem call that needs to be executed by VMs message.
+/// The corresponding value needs to be aligned with the specification.
+#[non_exhaustive]
+pub enum Opcode {
+    Init = 26,
+}
+
+impl TryFrom<u32> for Opcode {
+    type Error = ();
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            26 => Ok(Opcode::Init),
+            _ => Err(()),
+        }
+    }
+}
+
+/// InHeader represents the incoming message header in the filesystem call.
+#[repr(C)]
+#[derive(Debug, Default, Clone, Copy)]
+pub struct InHeader {
```

Comment:
This is a pub struct, how about adding comments for every field?

Reply:
I haven't found any official documentation describing what each field here means, and now each field here is just a reference to virtiofsd.

In addition, all structs here will only be used internally and will not be exposed to users. I'll add a link to virtiofsd, or I can add explanations for the fields I use.

Reply:
> In addition, all structs here will only be used internally and will not be exposed to users.

Hi, I'd like comments here to make our lives easier. As time passes, we might forget what these fields mean. Leaving comments can save readers a lot of time by reducing guesswork.

> I'll add a link to virtiofsd, or I can add explanations for the fields I use.

Adding a link to virtiofsd is enough. And adding explanations for used fields will be very helpful.